documentation/en/api-experimental-user-facing-v2-docs/api-v2-experimental.md

```diff
@@ -0,0 +1,1219 @@
+
+### Request Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "method": "Filecoin.StateGetID",
+  "params": [
+    "f2kc5mnx4pvgdpm5lzgutbfuzcnmkedlgd5wvyky", // A robust address
+    {
+      "tag": "latest" // Resolve based on the latest chain state
+    }
+  ],
+  "id": 1
+}
+```
+
+This example resolves the given robust address to its ID address at the latest tipset.
+
+### Response
+
+The method returns the ID address (string) corresponding to the input address at the specified tipset.
+
+**JSON-RPC Response Example:**
+
+```json
+{
+  "jsonrpc": "2.0",
+  "result": "f01234", // The resolved ID address
+  "id": 1
+}
+```
+
+If the address cannot be resolved to an ID address at the specified tipset state (e.g., it never existed or was created after that tipset), an error is returned.
+
+# Consensus Protocol Notes
+
+The Filecoin network is currently in the process of testing and activating Filecoin Fast Finality (F3), a new consensus mechanism that provides faster finality guarantees. During this transition period, the network will contain a mix of nodes - some with F3 enabled and others still using only the traditional Expected Consensus (EC) finality mechanism.
+
+This heterogeneous environment creates potential challenges for applications that need consistent finality information. The V2 APIs address this by implementing graceful fallback mechanisms that ensure a smooth API experience regardless of a node's configuration. While EC finality will always be available as a reliable fallback, the APIs are designed to automatically use F3 when available, handling edge cases transparently for the application developer.
+This approach ensures that applications can be built today that will seamlessly benefit from improved finality as F3 adoption increases across the network, without requiring code changes or special handling of different node types.
+
+## Understanding `finalized` TipSets
+
+The `finalized` tag is a powerful concept in the V2 APIs that returns the tipset considered finalized by the node's consensus protocol. This enables applications to work with confirmed blockchain data without needing to understand the underlying finality mechanism.
+
+### Node Configuration Impact
+
+How the `finalized` tag behaves depends on the node's configuration:
+
+1. **When F3 is Enabled and Ready**:
+    - The API returns the tipset identified by the latest F3 certificate.
+    - This provides faster finality guarantees (typically within minutes).
+    - The finalized tipset comes directly from the consensus certificates.
+2. **When F3 is Disabled or Not Ready**:
+    - The API automatically falls back to EC finality.
+    - EC finality is based on chain depth (currently 900 epochs, or ~7.5 hours).
+    - The finalized tipset is calculated as: `heaviestTipSet.Height - policy.ChainFinality`.
+
+### Graceful Fallback Mechanism
+
+The implementation includes several fallback scenarios to ensure robust operation:
```

Comment:
![image](https://github.com/user-attachments/assets/c4402a8e-3866-4a8c-99aa-e16d32f6b67a)

Reply:
Doh - good catch.  Fixed in https://github.com/filecoin-project/lotus/pull/13051/commits/bbd32817386a869a0c0df367e7fb75fb3ba1059b.  Resolving.